### PR TITLE
Test nodereplacement with multiple osd ids

### PR DESCRIPTION
**Pull request by yitzhak12**

### CODE REVIEW

File: ocs_ci/ocs/resources/pod.py [hunk 1]
```diff
@@ -1761,24 +1761,25 @@ def check_toleration_on_pods(toleration_key=constants.TOLERATION_KEY):
...
 
 
-def run_osd_removal_job(osd_id):
+def run_osd_removal_job(osd_ids=None):
     """
     Run the ocs-osd-removal job
 
     Args:
-        osd_id (str): The osd id
+        osd_ids (list): The osd ids.
```
Comment (AviadP) [inline ocs_ci/ocs/resources/pod.py:1769]:
```suggestion
        osd_ids (list): The osd IDs.
```

**Comment by yitzhak12** [comment]
The last PR validation https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/4188/ is passed. The error is in the sanity check and not in the test. I am checking now what the issue is.

File: ocs_ci/ocs/node.py [hunk 2]
```diff
@@ -982,9 +983,19 @@ def delete_and_create_osd_node_vsphere_upi_lso(osd_node_name, use_existing_node=
...
+    assert osd_ids, f"The node {osd_node_name} does not have osd pods"
+
+    ocs_version = config.ENV_DATA["ocs_version"]
+    if len(osd_ids) > 1 and Version.coerce(ocs_version) <= Version.coerce("4.6"):
+        log.warning(
+            f"We have {len(osd_ids)} osd ids, and ocs version is {ocs_version}. "
+            f"The ocs-osd-removal job works with multiple ids only from ocs version 4.7"
+        )
+        assert False
+
```
Comment (ebenahar) [inline ocs_ci/ocs/node.py:996]:
```suggestion
    ocs_version = config.ENV_DATA["ocs_version"]
    assert len(osd_ids) > 1 and Version.coerce(ocs_version) <= Version.coerce("4.6"), (
        f"We have {len(osd_ids)} osd ids, and ocs version is {ocs_version}. "
        f"The ocs-osd-removal job works with multiple ids only from ocs version 4.7"
    )

```

File: ocs_ci/ocs/node.py [hunk 3]
```diff
@@ -1012,25 +1026,27 @@ def delete_and_create_osd_node_vsphere_upi_lso(osd_node_name, use_existing_node=
     )
     assert res, "Failed to add the new node to LVD and LVS"
 
-    log.info("Verify new pv is available...")
-    is_new_pv_available = verify_new_pv_available_in_sc(old_pv_objs, sc_name)
-    assert is_new_pv_available, "New pv is not available"
+    log.info("Verify new pvs is available...")
```
Comment (ebenahar) [inline ocs_ci/ocs/node.py:1029]: replace `is` with `are`

File: ocs_ci/ocs/node.py (hunk 2 above)
Comment (yitzhak12) [reply ocs_ci/ocs/node.py:996]: I think you meant `assert not (
        len(osd_ids) > 1 and Version.coerce(ocs_version) <= Version.coerce("4.6")
    ), (
        f"We have {len(osd_ids)} osd ids, and ocs version is {ocs_version}. "
        f"The ocs-osd-removal job works with multiple ids only from ocs version 4.7"
    )`. We need to exclude this condition

Comment (ebenahar) [reply ocs_ci/ocs/node.py:996]: thanks for checking. It's ok to keep it as you did initially

File: ocs_ci/ocs/node.py (hunk 3 above)
Comment (yitzhak12) [reply ocs_ci/ocs/node.py]: Fixed

File: ocs_ci/ocs/node.py (hunk 2 above)
Comment (yitzhak12) [reply ocs_ci/ocs/node.py]: ok, np

**Review by ebenahar — APPROVED** [review]

**Comment by yitzhak12** [comment]
The last PR validation job https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/4247/ failed at the sanity check at the end because the noobaa health check failed. I am trying to figure out what caused this problem, but anyway, the test passed so I am marking it as verified.

File: ocs_ci/ocs/node.py [hunk 4]
```diff
@@ -998,9 +1009,12 @@ def delete_and_create_osd_node_vsphere_upi_lso(osd_node_name, use_existing_node=
     assert new_node_name, "Failed to create a new node"
     log.info(f"New node created successfully. Node name: {new_node_name}")
 
+    num_of_new_pvs = len(osd_ids)
+    log.info(f"Number of the expected new pvs = {num_of_new_pvs}")
     # If we use LSO, we need to create and attach a new disk manually
     new_node = get_node_objs(node_names=[new_node_name])[0]
-    add_disk_to_node(new_node)
+    for i in range(num_of_new_pvs):
+        add_disk_to_node(new_node)
```
[the remark this answers is not in the document]
Comment (yitzhak12) [reply ocs_ci/ocs/node.py:1017]: Yes. The purpose here is to add the number of disks as were in the previous node - so we can have the same amount of PVs. Maybe it should be explained in more detail...

**Review by OdedViner — APPROVED** [review]
LGTM

**Review by AviadP — APPROVED** [review]

**Pull request merged**